From 7871f71ab8c503967ea6d9538e4e5bd8eb59594b Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Mon, 8 Aug 2016 18:24:50 -0400
Subject: [PATCH 0725/1722] drm/amd/dal: skip underlay pipe for backend
 programming

This change eanbles hotplug and mode change during mpo

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c                    | 2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index 3b14f56..c0c535c 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -520,7 +520,7 @@ static void program_timing_sync(
         struct pipe_ctx *unsynced_pipes[MAX_PIPES] = { NULL };
         
 	for (i = 0; i < pipe_count; i++) {
-		if (!ctx->res_ctx.pipe_ctx[i].stream)
+                if (!ctx->res_ctx.pipe_ctx[i].stream || ctx->res_ctx.pipe_ctx[i].top_pipe)
 			continue;
         }
  
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index 9775902..8a24658 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1268,7 +1268,7 @@ static void switch_dp_clock_sources(
 	for (i = 0; i < MAX_PIPES; i++) {
 		struct pipe_ctx *pipe_ctx = &res_ctx->pipe_ctx[i];
 
-		if (pipe_ctx->stream == NULL)
+		if (pipe_ctx->stream == NULL || pipe_ctx->top_pipe)
 			continue;
 
 		if (dc_is_dp_signal(pipe_ctx->stream->signal)) {
@@ -1386,7 +1386,7 @@ static enum dc_status apply_ctx_to_hw(
 		struct pipe_ctx *pipe_ctx = &context->res_ctx.pipe_ctx[i];
 		struct dc_bios *dcb;
 
-		if (pipe_ctx->stream == NULL)
+		if (pipe_ctx->stream == NULL || pipe_ctx->top_pipe)
 			continue;
 
 		if (pipe_ctx->stream == pipe_ctx_old->stream) {
-- 
2.7.4

